scripts/src/logging.ts

```diff
@@ -0,0 +1,84 @@
+import * as winston from "winston";
+import { GenericTextTransportOptions } from "winston";
+import { GenericTransportOptions } from "winston";
+import { FileTransportOptions } from "winston";
+
+export interface LogTarget {
+	name: string;
+	type: "console" | "file";
+	level: "debug" | "info" | "warn" | "error";
+	format?: "string" | "json" | "pretty-json"; // default to "string"
+	timestamp?: boolean | (() => string | boolean);
+}
+
+export interface ConsoleTarget extends LogTarget {
+	type: "console";
+	name: "console";
+}
+
+export interface FileTarget extends LogTarget {
+	type: "file";
+	file: string;
+}
+
+let defaultLogger: winston.LoggerInstance;
+
+export function initLogger(...targets: LogTarget[]): winston.LoggerInstance {
+	const options: winston.LoggerOptions = {};
+	options.transports = targets.map(target => createTarget(target));
+	defaultLogger = new winston.Logger(options);
+	return defaultLogger;
+}
+
+export function getDefaultLogger(): winston.LoggerInstance {
+	return defaultLogger;
+}
+
+type WinstonTransportOptions = GenericTransportOptions & GenericTextTransportOptions & { stringify?: boolean };
+function createTarget(target: LogTarget): winston.TransportInstance {
+	let cls: { new (options: WinstonTransportOptions): winston.TransportInstance };
+	const defaults: WinstonTransportOptions = {
+		timestamp: true,
+	};
+	const options: WinstonTransportOptions = {
+		level: target.level,
+		timestamp: target.timestamp,
+	};
+
+	if (target.format === "json" || target.format === "pretty-json") {
```

Comment:
You can do:
`if (["json", "pretty-json"].includes(target.format)) {`
It's shorter, more readable and easier to add and remove items